src/main/kotlin/com/ampnet/auditornode/service/impl/GethEthereumRpc.kt

```diff
@@ -38,7 +38,10 @@ class GethEthereumRpc @Inject constructor(private val web3j: Web3j) : EthereumRP
                 " data: ${web3jTransaction.data})"
         }
 
-        val result = web3j.ethCall(web3jTransaction, DefaultBlockParameter.valueOf(block))?.send()?.value
+        val blockParameter = block.toBigIntegerOrNull()
+            ?.let { DefaultBlockParameter.valueOf(it) }
+            ?: DefaultBlockParameter.valueOf(block)
```

Comment:
```suggestion
block.toBigIntegerOrNull(16)?.let { DefaultBlockParameterNumber(it) }
                ?: DefaultBlockParameterName. valueOf(blockParameter)
```
If the input parameter is in hexadecimal you should put base as 16.